cosmos-tx/src/msg.rs

```diff
@@ -26,3 +28,48 @@ impl From<Msg> for Any {
         msg.0
     }
 }
+
+/// Message types that can be converted to/from a [`Msg`].
+pub trait MsgType {
+    /// Attempt to parse this value from a [`Msg`].
+    fn from_msg(msg: &Msg) -> Result<Self>
+    where
+        Self: Sized;
+
+    /// Serialize this value as a [`Msg`].
+    fn to_msg(&self) -> Result<Msg>;
+}
+
+/// Proto types which can be used as a [`Msg`].
+pub trait MsgProto: Default + MessageExt {
+    /// Type URL value
+    const TYPE_URL: &'static str;
+}
```

Comment:
Not sure if there are other ready-made patterns for `prost_types::Any`, but this seemed reasonable enough to me given the use case.

Reply:
Guess not, and it looks like this might all change soon: https://github.com/cosmos/cosmos-sdk/issues/9063